hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/rollback/SparkMarkerBasedRollbackStrategy.java

```diff
@@ -91,4 +96,40 @@ public SparkMarkerBasedRollbackStrategy(HoodieTable<T, JavaRDD<HoodieRecord<T>>,
         FSUtils.getPartitionPath(config.getBasePath(), partitionPathStr), fileId, HoodieFileFormat.HOODIE_LOG.getFileExtension(), baseCommitTime)
         .collect(Collectors.toMap(HoodieLogFile::getFileStatus, value -> value.getFileStatus().getLen()));
   }
+
+  /**
+   * Gets all marker file paths
+   *
+   * @param instant
+   * @param parallelism
+   * @return
+   * @throws IOException
+   */
+  private List<String> getAllMarkerFilePaths(String instant, int parallelism) throws IOException {
+    String markerDir = table.getMetaClient().getMarkerFolderPath(instant);
+    FileSystem fileSystem = FSUtils.getFs(markerDir, context.getHadoopConf().newCopy());
+    Option<MarkerType> markerTypeOption = MarkerUtils.readMarkerType(fileSystem, markerDir);
+
+    if (!markerTypeOption.isPresent()) {
+      WriteMarkers writeMarkers = WriteMarkersFactory.get(MarkerType.DIRECT, table, instant);
+      return new ArrayList<>(writeMarkers.allMarkerFilePaths());
+    }
+
+    switch (markerTypeOption.get()) {
+      case DIRECT:
+        WriteMarkers writeMarkers = WriteMarkersFactory.get(MarkerType.DIRECT, table, instant);
+        return new ArrayList<>(writeMarkers.allMarkerFilePaths());
+      case TIMELINE_SERVER_BASED:
+        // Reads all markers written by the timeline server
+        Map<String, Set<String>> markersMap =
+            MarkerUtils.readTimelineServerBasedMarkersFromFileSystem(
+                markerDir, fileSystem, context, parallelism);
+        List<String> markers = new ArrayList<>();
+        markersMap.forEach((key, value) -> markers.addAll(value));
```

Comment:
can we do this in one line? 
markersMap.values(). .. collect() or something.

Reply:
Changed to this:
```
return markersMap.values().stream().flatMap(Collection::stream)
            .collect(Collectors.toCollection(ArrayList::new));
```